ush/mesoscale/evs_sref_precip.sh

```diff
@@ -31,22 +33,41 @@ for  obsv in ccpa ; do
   #***********************************************
   # Get prepbufr data files for validation
   #***********************************************
-  $USHevs/mesoscale/evs_prepare_sref.sh $obsv 
-  export err=$?; err_chk
+  if [ ! -d $COMOUTrestart/ccpa.$vday ] ; then
+    $USHevs/mesoscale/evs_prepare_sref.sh $obsv 
+    export err=$?; err_chk
+  else
+    #restart from previously existing ccpa files
+    cp -r $COMOUTrestart/ccpa.$vday $WORK
+  fi
 
   if [ $obsv = ccpa ] ; then
+    #restart for sref_apcp06 is set within evs_prepare_sref.sh
     $USHevs/mesoscale/evs_prepare_sref.sh  sref_apcp06
     export err=$?; err_chk
-    $USHevs/mesoscale/evs_prepare_sref.sh sref_apcp24_mean
-    export err=$?; err_chk
+
+    #if ${COMOUTfinal}/apcp24mean exists, is will be used in the restart next time 
+    if [ ! -d ${COMOUTfinal}/apcp24mean ] ; then 
+      $USHevs/mesoscale/evs_prepare_sref.sh sref_apcp24_mean
+      export err=$?; err_chk
+    fi
   fi
 
   #*******************************************************
   # Build sub-jobs
   #*****************************************************
   for fhr in fhr1 fhr2 ; do
   
-       >run_sref_mpi_${domain}.${obsv}.${fhr}.sh
+    >run_sref_mpi_${domain}.${obsv}.${fhr}.sh
+
+    #############################################################################################################
+    # Adding following "if blocks"  for restart capability: 
+    #  1. check if *.completed files for 5  METplus processes (gneensprod, ens, mean and prob) exist, respectively
+    #  2. if any of the 5 not exist, then run its METplus, then mark it completed for restart checking next time
+    #  3. if any one of the 5 exits, skip it. But for gneensprod, all of the nc files generated from previous run
```

Comment:
Thank you Binbin! I spotted one more typo that you can correct. See line 67, should be genensprod.